app/controllers/home_controller.rb

```diff
@@ -2,5 +2,6 @@ class HomeController < ApplicationController
   after_action :verify_authorized, except: [:index]
 
   def index
+    redirect_to current_user.after_sign_in_path if user_signed_in?
```

Comment:
I see. Then maybe an approach like this ? 
https://www.viget.com/articles/using-routing-constraints-to-root-your-app

> On Jul 18, 2016, at 7:03 AM, Erinna Chen <notifications@github.com> wrote:
> 
> In app/controllers/home_controller.rb:
> 
> > @@ -2,5 +2,6 @@ class HomeController < ApplicationController
> >    after_action :verify_authorized, except: [:index]
> >  
> >    def index
> > +    redirect_to current_user.after_sign_in_path if user_signed_in?
> @jedschneider the flow of this story has been: I log in as a user, and I am redirected to the correct path depending on if I am a student or teacher. That is controlled by devise. After I am signed in, I return to the root path, then currently I see a link to login, if I click this link, devise will say I'm already signed in and redirect afterwards. This redirect line is to bypass that and instead go to the compass or dashboard instead if I have already logged in.
> What do you think? Also, this controller does not inherit from any of the Devise controllers.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>